README.md

```diff
@@ -551,6 +551,71 @@ if p.Subcommand() == nil {
 }
 ```
 
+### Option groups
+
+Option groups are a hybrid between subcommands and embedded structs. Option
+groups create logical collections of related arguments with a help description,
+and can be embedded in other groups and subcommands. Option groups can combine
+configuration structs of multiple modules without requiring embedding.
+
+This can be implemented with `go-arg` as follows:
+
+```go
+type Repository struct {
+    URL      string       `arg:"--host" help:"URL of the repository" default:"docker.io"`
+    User     string       `arg:"--user,env:REPO_USERNAME" help:"username to connect as"`
+    Password string       `arg:"--,env:REPO_PASSWORD" help:"password to connect with"`
+}
+type BuildCmd struct {
+	Context  string
+	Tag      string
+}
+type PushCmd struct {
+	Repo     Repository  `arg:"group:Repository" help:"Optional repository options."`
+	Tag      string `help:"Tag"`
+}
+var args struct {
+	Build    *BuildCmd    `arg:"subcommand:build"`
+	Push     *PushCmd     `arg:"subcommand:push"`
+	Quiet    bool         `arg:"-q" help:"Quiet"` // this flag is global to all subcommands
+}
+
+arg.MustParse(&args)
+
+switch {
+case args.Build != nil:
+	fmt.Printf("build %s as %q\n", args.Build.Context, args.Build.Tag)
+case args.Push != nil:
+	fmt.Printf("push %q to %q\n", args.Push.Tag, args.Push.Repo.URL)
+}
+```
+
+The push command help message would look like:
+
+```text
+Usage: example push [--tag TAG] [--host HOST] [--user USER]
+
+Repository options:
+Optional repository options.
```

Comment:
This line seems repetitive after the previous line. Can the help-generation logic can be changed so that this doesn't appear?

Reply:
Updated in 2 ways:
1. the actual text is provided by the `help:"..."` of the repository option group. I've updated the example to be less repetitive.
2. the option group name (header) and the optional help text are now separated by a white line to improve readability

Example usage with help text for the option group:
```
...
Repository options:

Change the default registry to push to.

  --host HOST            URL of the repository [default: docker.io]
  ...
```

or without help text:
```
...
Repository options:

  --host HOST            URL of the repository [default: docker.io]
  ...
```